packages/client-proxy/internal/edge-pass-through/proxy.go

```diff
@@ -0,0 +1,217 @@
+	}
+
+	// AWS ALB health check does not allow us to do health check on different HTTP protocol that
+	// on that that is used for actually proxying. So we cannot use edge API for health check as in other cases.
+	if fullMethodName == grpcHealthMethodName {
+		// we don't want to directly return error when service is marked as unhealthy
+		//  state should be managed by load balancer that will stop sending requests to this service
+		if s.info.GetStatus() != api.Unhealthy {
+			return status.Error(codes.OK, "healthy")
+		}
+
+		return status.Error(codes.Unavailable, "unhealthy")
+	}
+
+	// We require that the director's returned context inherits from the serverStream.Context().
+	backendConn, err := s.director(serverStream.Context())
+	if err != nil {
+		return err
+	}
+
+	clientCtx, clientCancel := context.WithCancel(s.ctx)
+	defer clientCancel()
+
+	clientStream, err := grpc.NewClientStream(clientCtx, clientStreamDescForProxying, backendConn, fullMethodName)
+	if err != nil {
+		return err
+	}
+
+	// Explicitly *do not close* s2cErrChan and c2sErrChan, otherwise the select below will not terminate.
+	// Channels do not have to be closed, it is just a control flow mechanism, see
+	// https://groups.google.com/forum/#!msg/golang-nuts/pZwdYRGxCIk/qpbHxRRPJdUJ
+	s2cErrChan := s.forwardServerToClient(serverStream, clientStream)
+	c2sErrChan := s.forwardClientToServer(clientStream, serverStream)
+
+	// We don't know which side is going to stop sending first, so we need a select between the two.
+	for i := 0; i < 2; i++ {
+		select {
+		case s2cErr := <-s2cErrChan:
+			if s2cErr == io.EOF {
+				// this is the happy case where the sender has encountered io.EOF, and won't be sending anymore./
+				// the clientStream>serverStream may continue pumping though.
+				clientStream.CloseSend()
+			} else {
+				// however, we may have gotten a receive error (stream disconnected, a read error etc) in which case we need
+				// to cancel the clientStream to the backend, let all of its goroutines be freed up by the CancelFunc and
+				// exit with an error to the stack
+				clientCancel()
+				return status.Errorf(codes.Internal, "failed proxying s2c: %v", s2cErr)
+			}
+		case c2sErr := <-c2sErrChan:
+			// This happens when the clientStream has nothing else to offer (io.EOF), returned a gRPC error. In those two
+			// cases we may have received Trailers as part of the call. In case of other errors (stream closed) the trailers
+			// will be nil.
+			serverStream.SetTrailer(clientStream.Trailer())
+			// c2sErr will contain RPC error from client code. If not io.EOF return the RPC error as server stream error.
+			if c2sErr != io.EOF {
+				return c2sErr
+			}
+			return nil
+		}
+	}
+
+	return status.Errorf(codes.Internal, "gRPC proxying should never reach this stage.")
+}
+
+func (s *NodePassThrough) forwardClientToServer(src grpc.ClientStream, dst grpc.ServerStream) chan error {
```

Comment:
In case of one2one there's different implementation in the source code
https://github.com/siderolabs/grpc-proxy/blob/8614c7158032488d36285ce2245d06f49d7447c4/proxy/handler_one2one.go#L59

Reply:
maybe we don't even need client streaming right now, i would jsut reference the code for potential later use

Reply:
agree, using more simple version without stream support